integration/update_test.go

```diff
@@ -460,3 +461,228 @@ func TestUpdateSet(t *testing.T) {
 		})
 	}
 }
+
+func TestSetOnInsert(t *testing.T) {
+	t.Parallel()
+
+	notModified := &mongo.UpdateResult{
+		MatchedCount:  0,
+		ModifiedCount: 0,
+		UpsertedCount: 1,
+	}
+
+	for name, tc := range map[string]struct {
+		filter      bson.D
+		setOnInsert any
+		stat        *mongo.UpdateResult
+		res         bson.D
+		err         *mongo.WriteError
+		alt         string
+	}{
+		"doc": {
+			filter:      bson.D{{"_id", "doc"}},
```

Comment:
here and with `array` - some versions of MongoDB can have different answers, so I suggest keeping those tests. 
Don't you mind?

Reply:
#679

Previous MongoDB versions relied heavily on OP_QUERY messages that are deprecated in 5.0 and mostly not implemented by us. People just can't migrate from those versions to FerretDB without updating their expectations.

That case tests the edge case in the current behavior, and that's good. But we should not test older versions.